js/store.js

```diff
@@ -134,19 +256,74 @@ const store = {
     const firstParty = this.getWebsite(origin);
     const thirdParty = this.getWebsite(target);
 
+/*
+  if not {},
+
+  firstParty is expected to match this format:
+  {
+    thirdPartyHostnames: [
+      "www.thirdpartydomain.com",
+      ...
+    ]
+  }
+  thirdParty is expected to match this format:
+  {
+    firstPartyHostnames: [
+      "www.firstpartydomain.com",
+      ...
+    ],
+    isIgnored: true, (optional)
+    whiteListedFirstParty: "www.google.com" (optional)
+  }
+  */
+
+    let whiteListOverride = false;
+    let onWhiteList = false;
```

Comment:
I have these booleans for this bit of code further down `store.setThirdParty` to update the `thirdParty` website object based on the whitelist/override check done when the `firstParty website object was updated, so I don't have to perform the whitelist check again.


```javascript
if (!('firstPartyHostnames' in thirdParty)) {
      thirdParty['firstPartyHostnames'] = [];
    }
    if (!thirdParty['firstPartyHostnames'].includes(origin)) {
      if (whiteListOverride) {
        thirdParty['firstPartyHostnames']
          .push(thirdParty['whiteListedFirstParty']);
      }
      if (newThirdParty) {
        thirdParty['firstPartyHostnames'].push(origin);
      }
    }
```

It's possible with some refactoring of this method that I wouldn't need these booleans.